lib/pbench/cli/server/shell.py

```diff
@@ -25,19 +29,71 @@ def app():
     return create_app(server_config)
 
 
+def add_to_path(path_el: str, path: str) -> str:
+    """Convenience function to hide f-string syntax used to construct a new
+    PATH environment variable.
+
+    Arguments:
+
+        path_el - a new path element string to be pre-pended to `path`
+        path - the exiting PATH environment variable value
+
+    Returns:
+
+        A string representing the newly constructed PATH value
+    """
+    return f"{path_el}:{path}"
+
+
 def find_the_unicorn(logger: Logger):
-    if site.ENABLE_USER_SITE:
-        local = Path(site.getuserbase()) / "bin"
-        if (local / "gunicorn").exists():
-            # Use a `pip install --user` version of gunicorn
-            os.environ["PATH"] = str(local) + ":" + os.environ["PATH"]
-            logger.info(
-                "Found a local unicorn: augmenting server PATH to {}",
-                os.environ["PATH"],
-            )
+    """Add the location of the `pip install --user` version of gunicorn to the
+    PATH if it exists.
+    """
+    local = Path(site.getuserbase()) / "bin"
+    if (local / "gunicorn").exists():
+        # Use a `pip install --user` version of gunicorn
+        os.environ["PATH"] = add_to_path(local, os.environ["PATH"])
+        logger.info(
+            "Found a local unicorn: augmenting server PATH to {}",
+            os.environ["PATH"],
+        )
+
+
+def wait_for_postgresql(db_uri: str, timeout: int):
+    """Wait for PostgreSQL database server to become available.  While we
+    encounter "connection refused", sleep one second, and then try again.
+
+    The timeout argument to `create_connection()` does not play into the retry
+    logic, see:
+
+      https://docs.python.org/3.9/library/socket.html#socket.create_connection
+
+    Arguments:
+
+        timeout: integer number of seconds to wait before giving up attempts to
+                 connect to the database
+
+    Raises a ConnectionRefusedError on failure.
```

Comment:
I don't think this function actually raises `ConnectionRefusedError` on failure:  other exceptions (from `create_connection()` might propagate out, but `ConnectionRefusedError` is caught and handled, and, if the timeout expires, then the function returns normally (with no return value).  (Did you mean to raise an exception after line 86?)

Also, not to quibble or anything, but `timeout` really refers to the number of attempts -- with a 1 second delay between them -- that the function will make...which isn't quite the same thing as the "number of seconds to wait".  If you want to implement what you've described, then line 80 should be something like
```
    end = time.time() + timeout
    while time.time() <= end:
```
🤷